### PR TITLE
[Magic WAN] strongSwan and VyOS IPsec config templates

### PR DESCRIPTION
**Pull request by dcpena**
Added config templates for strongSwan and VyOS to address [PCX-3609](https://jira.cfops.it/browse/PCX-3609).

### CODE REVIEW

File: content/magic-wan/tutorials/ipsec/_index.md [hunk 1]
```diff
@@ -1,14 +1,33 @@
...
+
+To learn more about Anycast IPsec, refer to [What is IPsec?](https://www.cloudflare.com/learning/network-layer/what-is-ipsec/).
 
 Before you begin, make sure you already have an Account ID and API Key.
 
+## IPsec process
+
+Review the information below to learn more about phases for IPsec connections.
+
+ - **Negotiation or IKE phase 1**: Endpoints use IPsec to negotiate the set of policies used in the connection and has two modes: main and aggressive. Some solutions, like [EdgeConnect](/magic-wan/partners/aruba-edgeconnect), only offer an aggressive mode.
```
Comment (kodster28) [inline content/magic-wan/tutorials/ipsec/_index.md]:
```suggestion
 - **Negotiation or IKE phase 1**: Endpoints use IPsec to negotiate the set of policies used in the connection and has two modes: main and aggressive. Some solutions, like [EdgeConnect](/magic-wan/partners/aruba-edgeconnect/), only offer an aggressive mode. 
```

File: content/magic-wan/tutorials/ipsec/_index.md [hunk 2]
```diff
@@ -1,14 +1,33 @@
...
+
+ - **Negotiation or IKE phase 1**: Endpoints use IPsec to negotiate the set of policies used in the connection and has two modes: main and aggressive. Some solutions, like [EdgeConnect](/magic-wan/partners/aruba-edgeconnect), only offer an aggressive mode. 
+
+    {{<Aside type="note" header="Note:">}}
+
+  The IKE aggressive mode ensures security but does not allow room for negotiation. With the main mode, negotiations continue until a consensus is reached.
+
+    {{</Aside>}}
+
+- **IPSec circuit or IKE Phase 2**: An IPsec circuit is established based on the authentication and encryption methods agreed upon in IKE Phase 1. During this phase, authentication and encryption during the data transmission phase is determined, and random, cryptographic numbers are exchanged to authenticate sessions.
```
Comment (kodster28) [inline content/magic-wan/tutorials/ipsec/_index.md]:
```suggestion
- **IPSec circuit or IKE Phase 2**: An IPsec circuit is established based on the authentication and encryption methods agreed upon in IKE Phase 1. During this phase, authentication and encryption during the data transmission phase is determined and random, cryptographic numbers are exchanged to authenticate sessions.
```

File: content/magic-wan/tutorials/ipsec/cisco.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+---
+pcx-content-type: tutorial
+title: Cisco
+_build:
```
Comment (kodster28) [inline content/magic-wan/tutorials/ipsec/cisco.md:4]: ???
Should you just use a `hidden` property instead? or is this the best way?

File: content/magic-wan/tutorials/ipsec/strongswan.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
+---
```
Comment (kodster28) [inline content/magic-wan/tutorials/ipsec/strongswan.md:1]: This should match what we do for the others.

File: content/magic-wan/tutorials/ipsec/strongswan.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+pcx-content-type: tutorial
+title: strongSwan
+---
+
+# strongSwan
+
+This tutorial contains a sample template of the `ipsec.conf`file for a working IPsec tunnel configuration established between a Linux machine running strongSwan and Cloudflare’s Magic service.
```
Comment (kodster28) [inline content/magic-wan/tutorials/ipsec/strongswan.md]:
```suggestion
This tutorial contains a sample template of the `ipsec.conf` file for a working IPsec tunnel configuration established between a Linux machine running strongSwan and Cloudflare’s Magic service.
```

File: content/magic-wan/tutorials/ipsec/strongswan.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+    rightsubnet=0.0.0.0/0
+    rightauth=psk
+    ike=aes256gcm16-prfsha512-modp2048
+    esp=aes256gcm16-prfsha512-modp2048
+    replay_window=0
+```
+
+### Dead Peer Detection (DPD)
+
+In the above `ipsec.conf `file in `conn %default` section, setting `dpdaction=restart` enables Dead Peer Detection (DPD) to actively check and re-establish IPsec tunnels in the event of communication timeouts. In addition, `closeaction=restart` is set to actively re-establish the tunnels in the event that the remote peer (i.e. Cloudflare Magic service) unexpectedly closes it.
```
Comment (kodster28) [inline content/magic-wan/tutorials/ipsec/strongswan.md]:
```suggestion
In the above `ipsec.conf` file in `conn %default` section, setting `dpdaction=restart` enables Dead Peer Detection (DPD) to actively check and re-establish IPsec tunnels in the event of communication timeouts. In addition, `closeaction=restart` is set to actively re-establish the tunnels in the event that the remote peer (usually a Cloudflare Magic service) unexpectedly closes it.
```

File: content/magic-wan/tutorials/ipsec/vyos.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
+---
+pcx-content-type: tutorial
```
Comment (kodster28) [inline content/magic-wan/tutorials/ipsec/vyos.md:2]: same comment for tutorial

**Review by kodster28 — APPROVED** [review]

File: content/magic-wan/tutorials/ipsec/cisco.md (hunk 3 above)
Comment (dcpena) [reply content/magic-wan/tutorials/ipsec/cisco.md:4]: Chatted about this offline and resolving :)

File: content/magic-wan/tutorials/ipsec/_index.md [hunk 8]
```diff
@@ -1,14 +1,33 @@
...
-Use Anycast IPsec as an on-ramp to connect with your entire virtual network. With an IPsec tunnel, you can route traffic from your network to Cloudflare's edge and define static routes to direct traffic down the correct tunnel. To learn more about Anycast IPsec, refer to [What is IPsec?](https://www.cloudflare.com/learning/network-layer/what-is-ipsec/).
+Use Anycast IPsec as an on-ramp to connect with your entire virtual network. With an IPsec tunnel, you can route traffic from your network to Cloudflare's edge and define static routes to direct traffic down the correct tunnel. 
+
+To learn more about Anycast IPsec, refer to [What is IPsec?](https://www.cloudflare.com/learning/network-layer/what-is-ipsec/).
 
 Before you begin, make sure you already have an Account ID and API Key.
 
+## IPsec process
+
+Review the information below to learn more about phases for IPsec connections.
```
Comment (jlcftme) [inline content/magic-wan/tutorials/ipsec/_index.md]: should be "phases to _establish_ IPsec connections".

File: content/magic-wan/tutorials/ipsec/_index.md (hunk 1 above)
Comment (dpuga1) [reply content/magic-wan/tutorials/ipsec/_index.md]: I have no partners

**Pull request merged**